base/macros.hpp

```diff
@@ -162,6 +162,13 @@ inline void noreturn() { std::exit(0); }
 #  define TEMPLATE template
 #endif
 
+// Use libc++ optional if possible
+#if PRINCIPIA_COMPILER_MSVC
+#  define OPTIONAL_HEADER "optional.hpp"
+#else
+#  define OPTIONAL_HEADER <experimental/optional>
+#endif
+
```

Comment:
I hacked our version of `optional` so that it can be included as `<experimental/optional>` and I prepared #678 to fix Principia.  So this macro should not be needed anymore, which is a good thing because the linter is not in love with it.  It would be good if you could revert the definition and usages of `OPTIONAL_HEADER` (and then git pull, just to be sure).